packages/material-ui-lab/src/TreeItem/TreeItem.js

```diff
@@ -7,6 +7,7 @@ import Collapse from '@material-ui/core/Collapse';
 import { fade, withStyles, useTheme } from '@material-ui/core/styles';
 import { useForkRef } from '@material-ui/core/utils';
 import TreeViewContext from '../TreeView/TreeViewContext';
+import { DescendantProvider, useDescendant, useDescendantsInit } from '../TreeView/descendants';
```

Comment:
Using descendants allows us to get tree items in the correct order without reading children's props meaning we aren't so fragile and allow wrapped TreeItems.